### PR TITLE
[Style Conventions] Style Convention guidelines file added. Fixes #48

### LINKED ISSUE #48: [coding] Style conventions
we need a style convention section in which we describe our coding style conventions:

- commit message style (this is explained in the contributing guidelines but the section is not titled properly and I guess not many people are reading it)
- python code convention
- javascript code convention
- openwrt related code convention

### PR DESCRIPTION
**Pull request by HargovindArora**
This PR consists Style Conventions for various programming languages coding style. 
This PR intent to fix the issue #48. @nemesisdesign Please have a look at it. :)

### CODE REVIEW

File: developer/style-conventions.rst [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  of the OpenWISP's Python file.
+
+- To know more about flake visit `this page. <http://flake8.pycqa.org/en/latest/>`_.
+
+
+Javascript Code Conventions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- OpenWISP follows standard JavaScript coding style conventions that are generally
+  followed and accepted in most of the projects.
```
Comment (BrainBuzzer) [inline developer/style-conventions.rst]: OpenWISP follows standard coding style conventions or the ones that are defined in jslint file.

Comment (HargovindArora) [reply developer/style-conventions.rst]: Oh. Thanks, @BrainBuzzer for pointing it out. Ill fix this

**Comment by HargovindArora** [comment]
@BrainBuzzer I'll do it once I get a review from @nemesisdesign also! :)

File: developer/style-conventions.rst [hunk 2]
```diff
@@ -0,0 +1,120 @@
+Style Conventions
+=================
+
+We are thankful that you took your time to contribute to this community.
```
Comment (nemesisdesign) [inline developer/style-conventions.rst]: We already state this concept in other pages, therefore this line is not necessary, please remove it

Comment (nemesisdesign) [inline developer/style-conventions.rst]: We already state this concept in other pages, therefore this line is not necessary, please remove it

File: developer/style-conventions.rst [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+::
+
+  git pull origin master
+  # if there's an issue for your patch
+  git checkout -b issues/48
+  # if you prefer a descriptive name
+  git checkout -b autoregistration
+
+2. Commit and push
```
Comment (nemesisdesign) [inline developer/style-conventions.rst]: Please change this to `2. Commit message style guidelines`

File: developer/style-conventions.rst [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+Steps to Commit your Code or Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Our main development branch is master it's our central development
+branch.
+
+Your code should be pushed from a different branch and then it is
+merged into master after proper review from the project maintainers.
+
+1. Create a new branch
```
Comment (nemesisdesign) [inline developer/style-conventions.rst]: Please change this to `1. Branch naming guidelines`

File: developer/style-conventions.rst [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+- commits should be descriptive in nature, the message should
+  explain the nature of the change
+- make sure to follow the code style used in the module
+  you are contributing to
+- before committing and pushing the changes, test the code both manually
+  and automatically with the automated test suite if applicable
+- after pushing your branch code, make a pull-request of that
+  corresponding change of yours which should contain a descriptive
+  message and mention the issue number as suggested in the example above
+
```
Comment (nemesisdesign) [inline developer/style-conventions.rst]: remove blank line

File: developer/style-conventions.rst [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+  Python Code <https://www.python.org/dev/peps/pep-0008/>`_.
+
+- OpenWISP uses `flake8 <https://pypi.python.org/pypi/flake8>`_ as it's style guide
+  enforcement tool or source code checker.
+
+- Please make sure you follow this code style while making changes to any
+  of the OpenWISP's Python file.
+
+- To know more about flake visit `this page. <http://flake8.pycqa.org/en/latest/>`_.
+
```
Comment (nemesisdesign) [inline developer/style-conventions.rst]: remove blank line

File: developer/style-conventions.rst [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+  To know more about JSlint please see `this <https://www.jslint.com/help.html>`_.
+
+- Proper structure and indentations should be maintained in code and descriptive
+  function and variable name should be used so that code is more readable and
+  self explanatory.
+
+- Please follow this `JavaScript Style Guide and Coding Conventions
+  <https://www.w3schools.com/js/js_conventions.asp>`_ link for proper
+  explanation and wonderful examples.
+
```
Comment (nemesisdesign) [inline developer/style-conventions.rst]: remove blank line

File: developer/style-conventions.rst [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+Openwrt Related Code Convention
```
Comment (nemesisdesign) [inline developer/style-conventions.rst]: change `Openwrt` to `OpenWRT`

File: developer/style-conventions.rst [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+- For proper python coding conventions one should read `PEP 8 -- Style Guide for
+  Python Code <https://www.python.org/dev/peps/pep-0008/>`_.
+
+- OpenWISP uses `flake8 <https://pypi.python.org/pypi/flake8>`_ as it's style guide
+  enforcement tool or source code checker.
+
+- Please make sure you follow this code style while making changes to any
+  of the OpenWISP's Python file.
+
+- To know more about flake visit `this page. <http://flake8.pycqa.org/en/latest/>`_.
```
Comment (nemesisdesign) [inline developer/style-conventions.rst]: we also use isort as mentioned in http://openwisp.io/docs/developer/contributing.html

**Comment by nemesisdesign** [comment]
@HargovindArora what about:

"I would really prefer if this information was merged with the contributing page because this information is what contributors need to know".

The information being added in this pull request duplicates some information we have here: http://openwisp.io/docs/developer/contributing.html

This is not good. Why can't we simply merge this piece of information in the contributing page directly?

**Comment by HargovindArora** [comment]
@nemesisdesign In the issue #48  you mentioned that "not many people are reading the section because of the title" so I thought why not create a new page! I think I misunderstood it! :sweat_smile:  I will merge the information! :+1:

**Pull request merged**